File: Data Structures & Algorithms/longest-common-prefix/submission-1.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        if(strs.length == 1) return strs[0];
        String s = strs[0],k;
        StringBuilder ans = new StringBuilder();
        StringBuilder pp = new StringBuilder();
        for(int i = 1; i < strs.length; i++)
        {
            pp = new StringBuilder();
            for(int j = 0; j < Math.min(s.length(),strs[i].length()); j++)
            {
                if(s.charAt(j) == strs[i].charAt(j)) pp.append(s.charAt(j));
                else break;
            }
            if(i == 1) ans = pp;
            else if(ans.length() > pp.length()) ans = pp;
        }
        return ans.toString();
    }
}

File: Data Structures & Algorithms/merge-sorted-array/submission-1.cpp
class Solution {
public:
    void merge(vector<int>& n1, int m, vector<int>& n2, int n) {
        int mn = m + n;
        for(int i = m - 1; i >= 0; i--) n1[i + n] = n1[i];
        for(int i = 0; i < n; i++) n1[i] = 0;
        int i = n;
        int j = 0;
        int k = 0;
        while(i < mn && j < n)
        {
            if(n1[i] < n2[j]) n1[k++] = n1[i++];
            else n1[k++] = n2[j++];
        }
        while(i < mn) n1[k++] = n1[i++];
        while(j < n) n1[k++] = n2[j++];
    }
};